### PR TITLE
Added support for multiple folder deployment

### PR DESCRIPTION
**Pull request by manujuniris**

### CODE REVIEW

File: aeweb.js [hunk 1]
```diff
@@ -1,129 +1,212 @@
...
-        
-        
-        
-      
+
+        txn = tx
+            .build(argv.seed, 0)
+            .originSign(originPrivateKey)
+
+        archethic.sendTransaction(txn, argv.endpoint)
```
Comment (samuel-uniris) [inline aeweb.js:176]: You should wait the completion before to continue further. Othewise the transaction may not be validated yet

**Review by samuel-uniris — CHANGES_REQUESTED** [review]
Good overall.
The thing is done and worling.

But in addition to the comments, I also think you can leverage async/await to avoid callbacks hell and make code more readable. 
Moreover I will request better error handling to catch the potential failed promises.

**Comment by redDwarf03** [comment]
hello 
could explain in the readme.md how to create subfolder please

is it 
aeweb deploy-folder --seed=myseedphrase --endpoint=https://testnet.archethic.net --folder=Aeweb/images ?

**Comment by redDwarf03** [comment]
just for info, I tried to publish the wallet with this script and with the content of the zip file

```

aeweb generate-address --seed=archethicWalletTest --index=0

aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/assets
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/assets/buy
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/assets/fonts
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/assets/icons
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/assets/themes
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/fonts
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/packages
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/packages/cupertino_icons
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/packages/cupertino_icons/assets
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/packages/font_awesome_flutter
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/packages/font_awesome_flutter/lib
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=assets/packages/font_awesome_flutter/lib/fonts
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=canvaskit
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=canvaskit/profiling
aeweb deploy-folder --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --folder=icons


aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/favicon.png 
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/flutter_service_worker.js
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/main.dart.js
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/main.dart.js.map
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/manifest.json
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/version.json
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/index.html
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/version.json
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/.last_build_id
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/icons/Icon-192.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/icons/Icon-maskable-192.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/icons/Icon-maskable-512.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/icons/Icon-512.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/AssetManifest.json
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/NOTICES
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/FontManifest.json
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/packages/cupertino_icons/assets/CupertinoIcons.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/packages/font_awesome_flutter/lib/fonts/fa-solid-900.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/packages/font_awesome_flutter/lib/fonts/fa-regular-400.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/packages/font_awesome_flutter/lib/fonts/fa-brands-400.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/fonts/MaterialIcons-Regular.otf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/warning.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/privacyPolicy.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/PreviousSignature.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/Code.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/TimeStamp.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/Type.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/OriginSignature.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/Address.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/ProofOfWork.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/Content.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/ProofOfIntegrity.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/Version.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/PreviousPublicKey.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/Amount.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/To.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/txInfos/Signature.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/authentication.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/autoLock.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/themes.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/lock.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/languages.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/walletServiceTerms.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/oracle.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/digital-key.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/nft.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/logout.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/authentLaunch.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/arrow1.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/arrow2.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/chains.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/url.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/pin-code.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/help.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/encrypted.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/key-word.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/shuffle.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/microscope.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/terms-and-conditions.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/transferNft.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/faq.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/address-book.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/key-ring.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/barcode.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/send.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/chf.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/krw.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/huf.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/nok.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/pkr.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/aed.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/sar.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/ils.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/sek.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/sgd.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/gbp.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/jpy.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/dkk.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/myr.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/hkd.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/usd.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/rub.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/brl.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/clp.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/pln.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/kwd.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/inr.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/czk.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/cny.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/aud.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/mxn.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/eur.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/zar.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/twd.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/cad.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/nzd.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/btc.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/idr.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/try.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/thb.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/php.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/currency/ars.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/nl.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/ja.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/en.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/fr.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/es.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/id.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/de.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/country/it.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/nodes.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/writing.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/money-currency.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/icons/addNft.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-LightItalic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-Medium.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-BoldItalic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-Light.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-ExtraLight.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-Bold.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-MediumItalic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-BlackItalic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-SemiBold.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-ExtraLightItalic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-ExtraBold.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-Black.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-Regular.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-Italic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-SemiBoldItalic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/fonts/Montserrat-ExtraBoldItalic.ttf
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Quickswap.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Zebitex.svg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Probit.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Pancake.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Zamio.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Rubic.svg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Uniswap.svg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/buy/Bithumb.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/light/logo_alone.svg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/light/logo_alone.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/light/logo.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/light/logo.svg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/dark/logo_alone.svg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/dark/logo_alone.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/dark/background.jpg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/dark/logo.png
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/assets/assets/themes/dark/logo.svg
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/canvaskit/canvaskit.js
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/canvaskit/profiling/canvaskit.js
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/canvaskit/profiling/canvaskit.wasm
aeweb deploy-file --seed=archethicWalletTest --endpoint=https://testnet.archethic.net --index=0 --path=./web/canvaskit/canvaskit.wasm
`
[web.zip](https://github.com/archethic-foundation/aeweb-cli/files/8026920/web.zip)
`

**Comment by redDwarf03** [comment]
hello
I run my script again with your last updates.
i obtain my html page: https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=text/html
The source code is ok but it seems the assets don't work.

i have a question.
when I send for example few images, I have always the same url : 
https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=image/png 

how can I display an image from a website host on archethic blockchain alone in a browser ?

**Comment by manujuniris** [comment]
> hello I run my script again with your last updates. i obtain my html page: https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=text/html The source code is ok but it seems the assets don't work.
> 
> i have a question. when I send for example few images, I have always the same url : https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=image/png
> 
> how can I display an image from a website host on archethic blockchain alone in a browser ?

Hello
The last update supports single file deployment and multiple files deployment.

At first, run - sudo npm link

To deploy a single file - 
1. Generate address -
aeweb generate-address --seed=myseedphrase --index=0
2. Transfer funds to the generated address using faucet -
https://testnet.archethic.net/faucet
3. Deploy a file -
aeweb deploy-file --seed=myseedphrase --endpoint=https://testnet.archethic.net --file=./image.png

To deploy a folder -
1. Generate address -
aeweb generate-address --seed=myseedphrase --index=0
2. Transfer funds to the generated address using faucet -
https://testnet.archethic.net/faucet
3. Deploy a folder - 
aeweb deploy-folder --seed=myseedphrase --endpoint=https://testnet.archethic.net --folder=Aeweb

Note - 
Here, Aeweb is the name of your folder inside which you may have your files
For multi-files, the transaction hash generated depends on the seed you choose and the filename

**Comment by redDwarf03** [comment]
> > hello I run my script again with your last updates. i obtain my html page: https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=text/html The source code is ok but it seems the assets don't work.
> > i have a question. when I send for example few images, I have always the same url : https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=image/png
> > how can I display an image from a website host on archethic blockchain alone in a browser ?
> 
> Hello The last update supports single file deployment and multiple files deployment.
> 
> At first, run - sudo npm link
> 
> To deploy a single file -
> 
> 1. Generate address -
>    aeweb generate-address --seed=myseedphrase --index=0
> 2. Transfer funds to the generated address using faucet -
>    https://testnet.archethic.net/faucet
> 3. Deploy a file -
>    aeweb deploy-file --seed=myseedphrase --endpoint=https://testnet.archethic.net --file=./image.png
> 
> To deploy a folder -
> 
> 1. Generate address -
>    aeweb generate-address --seed=myseedphrase --index=0
> 2. Transfer funds to the generated address using faucet -
>    https://testnet.archethic.net/faucet
> 3. Deploy a folder -
>    aeweb deploy-folder --seed=myseedphrase --endpoint=https://testnet.archethic.net --folder=Aeweb
> 
> Note - Here, Aeweb is the name of your folder inside which you may have your files For multi-files, the transaction hash generated depends on the seed you choose and the filename

and have you fot an example to deploy a subfolder please

**Comment by manujuniris** [comment]
> > > hello I run my script again with your last updates. i obtain my html page: https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=text/html The source code is ok but it seems the assets don't work.
> > > i have a question. when I send for example few images, I have always the same url : https://testnet.archethic.net/api/last_transaction/00fb98ea696f2005a2a022a6f95fa20f8d191a055b4bfcd957a2c2635c1c3817f1/content?mime=image/png
> > > how can I display an image from a website host on archethic blockchain alone in a browser ?
> > 
> > 
> > Hello The last update supports single file deployment and multiple files deployment.
> > At first, run - sudo npm link
> > To deploy a single file -
> > 
> > 1. Generate address -
> >    aeweb generate-address --seed=myseedphrase --index=0
> > 2. Transfer funds to the generated address using faucet -
> >    https://testnet.archethic.net/faucet
> > 3. Deploy a file -
> >    aeweb deploy-file --seed=myseedphrase --endpoint=https://testnet.archethic.net --file=./image.png
> > 
> > To deploy a folder -
> > 
> > 1. Generate address -
> >    aeweb generate-address --seed=myseedphrase --index=0
> > 2. Transfer funds to the generated address using faucet -
> >    https://testnet.archethic.net/faucet
> > 3. Deploy a folder -
> >    aeweb deploy-folder --seed=myseedphrase --endpoint=https://testnet.archethic.net --folder=Aeweb
> > 
> > Note - Here, Aeweb is the name of your folder inside which you may have your files For multi-files, the transaction hash generated depends on the seed you choose and the filename
> 
> and have you got an example to deploy a subfolder please ?

Yes, files inside the subfolder will also be deployed.
cli

File: aeweb.js [hunk 2]
```diff
@@ -1,129 +1,216 @@
...
+
+    },
+
+    handler: async function (argv) {
+        const content = fs.readFileSync(argv.file)
+        transaction = null
+        txBuilder = archethic.newTransactionBuilder("hosting")
+            .setContent(content)
+        const address = archethic.deriveAddress(argv.seed, 0)
+        const index = await archethic.getTransactionIndex(address, argv.endpoint)
```
Comment (samuel-uniris) [inline aeweb.js:101]: I think this call should be also in the try catch

File: aeweb.js [hunk 3]
```diff
@@ -1,129 +1,216 @@
...
-            
-            console.log(chalk.blue("Transaction Sent Successfully !"))
-            
-            console.log(chalk.green(argv.endpoint+"/api/last_transaction/"+(toHex(transaction.address))+"/content?mime="+mime.getType(argv.path)))
-            
-        })
+            const txBuilder = archethic.newTransactionBuilder("hosting")
+            txBuilder.setContent(content)
+
+            const index = await archethic.getTransactionIndex(address, argv.endpoint)
```
Comment (samuel-uniris) [inline aeweb.js:193]: Same here

**Review by samuel-uniris — CHANGES_REQUESTED** [review]

**Review by samuel-uniris — APPROVED** [review]

**Pull request merged**